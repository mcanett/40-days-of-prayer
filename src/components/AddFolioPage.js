import React from 'react';
import { connect } from 'react-redux';
import PartakerNameForm from './PartakerNameForm';
import { startAddFolio } from '../actions/partakers';
import * as routes from '../constants/routes';

export class AddFolioPage extends React.Component {
  onSubmit = (partakerName) => {
    this.props.startAddFolio(partakerName).then(() => {
      this.props.history.push(routes.FINANCES);
    });
  };
  render() {
    return (
      <div className="component">
        <h1 className="component__header">Creación de folio</h1>
        <PartakerNameForm
          onSubmit={this.onSubmit}
        />
      </div>
    );
  }
}

const mapDispatchToProps = (dispatch) => ({
  startAddFolio: (partakerName) => dispatch(startAddFolio(partakerName))
});

export default connect(undefined, mapDispatchToProps)(AddFolioPage);