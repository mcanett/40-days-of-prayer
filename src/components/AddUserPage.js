import React from 'react';
import { history } from '../routers/AppRouter';
import { startSignUp, saveUserInfo } from '../actions/auth';
import * as routes from '../constants/routes';

const AddUserPage = () => (
  <div>
    <h1>Registra un usuario</h1>
    <AddUserForm />
  </div>
);

const INITIAL_STATE = {
  username: '',
  email: '',
  passwordOne: '',
  passwordTwo: '',
  userType: '',
  error: null,
};

class AddUserForm extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      ...INITIAL_STATE
    }
  }

  handleChange = (e) => {
    const control = e.target.id;
    const value = e.target.value;
    this.setState(() => ({ [control]: value }));
  }

  onSubmit = (e) => {
    e.preventDefault();
    const {
      username,
      email,
      passwordOne,
      passwordTwo,
      userType
    } = this.state;

    startSignUp(email, passwordOne)
      .then(authUser => {
        this.setState({ ...INITIAL_STATE });
        saveUserInfo(authUser.uid, username, email, userType).then(() => {
          history.push(routes.USERS);
        });
      })
      .catch(error => {
        this.setState(() => ({ 'error': error }));
      });

  };

  render() {
    const {
      username,
      email,
      passwordOne,
      passwordTwo,
      userType,
      error,
    } = this.state;

    const isInvalid =
    passwordOne !== passwordTwo ||
    passwordOne === '' ||
    email === '' ||
    username === '' ||
    userType === '';
    
    return (
      <div>
        { error && <p style={{color: 'red'}}>{error.message}</p> }
        <form onSubmit={this.onSubmit}>
          <input
            id={"username"}
            value={username}
            onChange={this.handleChange}
            type="text"
            placeholder="Nombre completo"
          />
          <input
            id={"email"}
            value={email}
            onChange={this.handleChange}
            type="text"
            placeholder="Correo"
          />
          <input
            id={"passwordOne"}
            value={passwordOne}
            onChange={this.handleChange}
            type="password"
            placeholder="Contraseña"
          />
          <input
            id={"passwordTwo"}
            value={passwordTwo}
            onChange={this.handleChange}
            type="password"
            placeholder="Confirma contraseña"
          />
          <select
            id="userType"
            value={userType}
            onChange={this.handleChange}
          >
            <option value="">-- Tipo de usuario --</option>
            <option value="registry">Registro</option>
            <option value="finances">Finanzas</option>
            <option value="supervisor">Supervisor</option>
            <option value="admin">Administrador</option>
          </select>
          <button disabled={isInvalid}>Registrar</button>
        </form>
      </div>
    );
  }
}

export default AddUserPage;