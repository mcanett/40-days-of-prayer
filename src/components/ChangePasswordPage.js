import React from 'react';
import { history } from '../routers/AppRouter';
import { startPasswordUpdate } from '../actions/auth';
import * as routes from '../constants/routes';

const ChangePasswordPage = () => (
  <PasswordChangeForm />
);


const byPropKey = (propertyName, value) => () => ({
  [propertyName]: value
});

const INITIAL_STATE = {
  passwordOne: '',
  passwordTwo: '',
  error: null,
};

class PasswordChangeForm extends React.Component {
  constructor(props) {
    super(props);

    this.state = { ...INITIAL_STATE };
  };

  handleChange = (e) => {
    const control = e.target.id;
    const value = e.target.value;
    this.setState(byPropKey(control, value));
  };

  onSubmit = (e) => {
    e.preventDefault();
    const { passwordOne } = this.state;

    startPasswordUpdate(passwordOne)
      .then(() => {
        this.setState({ ...INITIAL_STATE });
        history.push(routes.PRIVATE_WELCOME);
      })
      .catch(error => {
        this.setState(byPropKey('error', error));
      });
  };

  render() {
    const {
      passwordOne,
      passwordTwo,
      error,
    } = this.state;

    const isInvalid =
      passwordOne !== passwordTwo ||
      passwordOne === '';

    return (
      <div className="component">
        { error && <p style={{color: 'red'}}>{error.message}</p> }
        <form onSubmit={this.onSubmit}>
          <input
            id="passwordOne"
            value={passwordOne}
            onChange={this.handleChange}
            type="password"
            placeholder="Nueva Contraseña"
          />
          <input
            id="passwordTwo"
            value={passwordTwo}
            onChange={this.handleChange}
            type="password"
            placeholder="Confirma Nueva Contraseña"
          />
          <button disabled={isInvalid} type="submit">
            Cambiar contraseña
          </button>

        </form>
      </div>
    );
  };
}


export default ChangePasswordPage;