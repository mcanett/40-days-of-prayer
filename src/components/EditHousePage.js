import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { BlueEditIcon } from './Icons';
import { getHouseRemainingCapacity } from '../utils/hostHelper';
import HousePartakerList from './HousePartakerList';

export class EditHousePage extends React.Component {
  constructor(props) {
    super(props);
  }

  render() { 
    const { host, facilitator, housePartakers, houseRemainingCapacity } = this.props;

    const { firstName: nameH, lastName: lastH, mothersSurname: mothersH } = host.name;
    const { streetName, houseNumber, neighborhood, zipCode } = host.hostInfo.address;

    const { firstName: nameF, lastName: lastF, mothersSurname: mothersF } = facilitator ? facilitator.name : {undefined, undefined, undefined};

    let facilitatorRender;
    if (facilitator && host === facilitator) {
      facilitatorRender = <div className="grid">
      <div className="cell__header cell__s">Facilitador:</div>
        <div className="cell__content cell__l">
          <h3>
            {nameF} {lastF} {mothersF}
          </h3>
        </div>
      </div>
    } else if (facilitator) {
      facilitatorRender = <div className="grid">
      <div className="cell__header cell__s">Facilitador:</div>
        <div className="cell__content cell__l">
          <h3>
            {nameF} {lastF} {mothersF}
          </h3>
        </div>
        <div className="cell__content cell__m">
          <h3>
            {facilitator.phone}
          </h3>
        </div>
        <div className="cell__content cell__xs">
          <h3>
            <Link to={{
              pathname: `/edit/${facilitator.id}`,
              search: '',
              state: { detail: 'hf' }
            }}><BlueEditIcon /></Link>
          </h3>
        </div>
      </div>
    } else {
      facilitatorRender = <div className="grid">
        <div className="cell__header cell__s">Facilitador:</div>
        <div className="cell__content cell__l">
          <h3>
            Esta casa no cuenta con facilitador
          </h3>
        </div> 
      </div>
    }



    return (
      <div className="component">
        <div className="component__container">
          <h1 className="component__header">Detalles de casa</h1>
          <div className="grid">  
            <div className="cell__header cell__xs">#:</div>
            <div className="cell__content cell__xs">
              <h3>
                {host.hostInfo.numberLabel}
              </h3>
            </div>
            <div className="cell__header cell__s">Horario:</div>
            <div className="cell__content cell__s">
              <h3>
                {host.hostInfo.houseSchedule}
              </h3>
            </div>
            <div className="cell__header cell__s">Capacidad:</div>
            <div className="cell__content cell__xs">
              <h3>
                { host.hostInfo.houseCapacity}
              </h3>
            </div>
            <div className="cell__header cell__s">Inscritos:</div>
            <div className="cell__content cell__xs">
              <h3>
                { host.hostInfo.houseCapacity - houseRemainingCapacity }
              </h3>
            </div>
          </div>
          
          <div className="grid">  
            <div className="cell__header cell__s">Dirección:</div>
            <div className="cell__content cell__xl">
              <h3>
                {streetName} {houseNumber}, {neighborhood} {zipCode}
              </h3>
            </div>
          </div>
          
          <div className="grid">  
            <div className="cell__header cell__s">Anfitrión:</div>
            <div className="cell__content cell__l">
              <h3>
                {nameH} {lastH} {mothersH}
              </h3>
            </div>
            <div className="cell__content cell__m">
              <h3>
                {host.phone}
              </h3>
            </div>
            <div className="cell__content cell__xs">
              <h3>
                <Link to={{
                  pathname: `/edit/${host.id}`,
                  search: '',
                  state: { detail: 'hf' }
                }}><BlueEditIcon /></Link>
              </h3>
            </div>
            </div>        
            {facilitatorRender}
          <h2 className="component__header">Participantes</h2>
          <HousePartakerList partakers={housePartakers}/>
        </div>
      </div>
    );
  }
}

const mapStateToProps = (state, props) => {
  const host = state.partakers.find((host) => host.id === props.match.params.id);

  const facilitatorSearch = state.partakers.filter((partaker) => partaker.houseId === host.id && partaker.facilitatorInfo !== undefined);
  const facilitator = host.facilitatorInfo ? host : facilitatorSearch ? facilitatorSearch[0] :  undefined;
  //const facilitator = facilitatorSearch ? facilitatorSearch[0] : undefined;

  const housePartakers = state.partakers.filter((partaker) => partaker.houseId === host.id && partaker.facilitatorInfo === undefined);

  return {
    host,
    facilitator,
    housePartakers,
    houseRemainingCapacity: getHouseRemainingCapacity(host, state.partakers),
  };
};

export default connect(mapStateToProps, undefined)(EditHousePage);