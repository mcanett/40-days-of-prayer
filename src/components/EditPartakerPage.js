import React from 'react';
import { connect } from 'react-redux';
import HostFacilitatorForm from './HostFacilitatorForm';
import PartakerBasicForm from './PartakerBasicForm';
import { startEditPartaker, startRemovePartaker } from '../actions/partakers';
import selectHosts from '../selectors/hosts';
import * as routes from '../constants/routes';

export class EditPartakerPage extends React.ComponentÂ {
  onSubmit = (partaker) => {
    this.props.startEditPartaker(this.props.partaker.id, partaker, this.props.userName).then(() => {
      this.props.history.push(routes.REGISTRY);
    });
  };
  onRemove = () => {
    this.props.startRemovePartaker(this.props.partaker.id);
    this.props.history.push(routes.REGISTRY);
  };

  render() {
    return (
      <div className="component">
      { this.props.location.state.detail === 'hf' ?
        <HostFacilitatorForm
          hosts={this.props.hosts}
          partaker={this.props.partaker}
          onSubmit={this.onSubmit}
        />
        :
        <PartakerBasicForm 
          hosts={this.props.hosts}
          partaker={this.props.partaker}
          onSubmit={this.onSubmit}
        />
      }
        {/*<button
          className="button button__danger"
          onClick={this.onRemove}
        >Eliminar Participante</button>*/}
      </div>
    );
  }
}

const mapStateToProps = (state, props) => ({
  partaker: state.partakers.find((partaker) => partaker.id === props.match.params.id),
  hosts: selectHosts(state.partakers),
  userName: state.auth.userName
});

const mapDispatchToProps = (dispatch) => ({
  startEditPartaker: (id, partaker, userName) => dispatch(startEditPartaker(id, partaker, userName)),
  startRemovePartaker: (id) => dispatch(startRemovePartaker({ id }))
});

export default connect(mapStateToProps, mapDispatchToProps)(EditPartakerPage);