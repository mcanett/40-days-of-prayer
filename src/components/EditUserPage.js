import React from 'react';
import { connect } from 'react-redux';
import { startEditUser } from '../actions/users';
import * as routes from '../constants/routes';

export class EditUserPage extends React.ComponentÂ {
  constructor(props) {
    super(props);

    this.state = {
      userName: props.user.userName,
      userType: props.user.userType,
      error: null,
    }
  }

  handleChange = (e) => {
    const control = e.target.id;
    const value = e.target.value;
    this.setState(() => ({ [control]: value }));
  }

  onSubmit = (e) => {
    e.preventDefault();

    const {
      userName,
      userType
    } = this.state;

    this.props.startEditUser(this.props.user.id, {userName, userType}).then(() => {
      this.props.history.push(routes.USERS);
    }).catch((e) => {
      this.setState(() => ({
        error: e
      }))
    });
  };

  render() {
    return (
      <div className="component">
        <div className="component__container">
          <h1 className="component__title">Editando usuario</h1>
          { this.state.error && <p style={{color: 'red'}}>{this.state.error.message}</p> }
          <form onSubmit={this.onSubmit}>
            <div className="input-group">
              <input 
                id="userName"
                className="text-input"
                value={this.state.userName}
                placeholder="Nombre usuario"
                onChange={this.handleChange}
              />
            </div>
            <div className="input-group">
              <select
                id="userType"
                className="select"
                value={this.state.userType}
                onChange={this.handleChange}
              >
                <option value="">-- Tipo de usuario --</option>
                <option value="registry">Registro</option>
                <option value="finances">Finanzas</option>
                <option value="supervisor">Supervisor</option>
                <option value="admin">Administrador</option>
              </select>
            </div>
            <div className="input-group">
              <button className="button button__positive">Guardar usuario</button>
            </div>
          </form>
        </div>
      </div>
    );
  }
}

const mapStateToProps = (state, props) => ({
  user: state.users.find((user) => user.id === props.match.params.id)
});

const mapDispatchToProps = (dispatch) => ({
  startEditUser: (id, user) => dispatch(startEditUser(id, user))
});

export default connect(mapStateToProps, mapDispatchToProps)(EditUserPage);

