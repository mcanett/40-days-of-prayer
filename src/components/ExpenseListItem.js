import React from 'react';
import { Link } from 'react-router-dom';

const ExpenseListItem = ({ count, id, description, amount, createdAt}) => (
  <div>
    <Link to={`edit/${id}`}>
      <h3>{count}. {description}</h3>
    </Link>
    <p>{amount} - {createdAt}</p>
    
  </div>
);

export default ExpenseListItem;