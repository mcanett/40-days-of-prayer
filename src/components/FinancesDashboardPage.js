import React from 'react';
import { NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
import FolioModal from './FolioModal';
import FinancesList from './FinancesList';
import FinancesListFilters from './FinancesListFilters';
import FinancesSummary from './FinancesSummary';
import { removeLastPartaker } from '../actions/partakers';Â 

export class FinancesDashboardPage extends React.Component {
  state = {
    lastPartaker: this.props.lastPartaker
  };

  handleClearCreatedFolio = () => {
    this.props.removeLastPartaker();
    this.setState(() => ({
      lastPartaker: null
    }));
	}

  render() {
    return (
      <div>
        <div>
          <NavLink to="/create/folio" activeClassName="is-active">Crear Folio</NavLink>
        </div>
        <div>
          <FinancesSummary />
          <FinancesListFilters />
          <FinancesList />
        </div>
        <FolioModal
            openModal={!!this.state.lastPartaker}
            lastPartaker={this.state.lastPartaker}
						handleClearCreatedFolio={this.handleClearCreatedFolio}
					/>
      </div>
    );;
  }
}

const mapStateToProps = (state) => ({
  lastPartaker: state.lastPartaker
});

const mapDispatchToProps = (dispatch) => ({
  removeLastPartaker: () => dispatch(removeLastPartaker())
});

export default connect(mapStateToProps, mapDispatchToProps)(FinancesDashboardPage);