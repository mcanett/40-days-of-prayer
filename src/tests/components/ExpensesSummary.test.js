import React from 'react';
import { shallow } from 'enzyme';
import {Â ExpensesSummary } from '../../components/ExpensesSummary';

test('should render a singular message for single expense', () => {
  const wrapper = shallow(
    <ExpensesSummary 
      expensesCount={1}
      expensesTotal={5000}
    />
  );
  expect(wrapper).toMatchSnapshot();
});

test('should render a plural message for multiple expenses', () => {
  const wrapper = shallow(
    <ExpensesSummary 
      expensesCount={3}
      expensesTotal={15000}
    />
  );
  expect(wrapper).toMatchSnapshot();
});

