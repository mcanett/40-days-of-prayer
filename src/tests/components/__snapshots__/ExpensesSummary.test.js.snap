// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a plural message for multiple expenses 1`] = `
<div>
  <h2>
    Viewing 
    3
     
    expenses
     totalling 
    $150.00
  </h2>
</div>
`;

exports[`should render a singular message for single expense 1`] = `
<div>
  <h2>
    Viewing 
    1
     
    expense
     totalling 
    $50.00
  </h2>
</div>
`;
